schemas/mozilla_nimbus_schemas/experiments/experiments.py

```diff
@@ -273,3 +294,50 @@ class NimbusExperiment(BaseModel):
         ),
         default=None,
     )
+
+    @model_validator(mode="after")
+    @classmethod
+    def validate_firefox_labs(cls, data: Self) -> Self:
+        if data.isFirefoxLabsOptIn:
+            if data.firefoxLabsTitle is None:
+                raise ValueError(
+                    "firefoxLabsTitle field cannot be none "
+                    "if isFirefoxLabsOptIn is True"
+                )
+            if data.firefoxLabsDescription is None:
+                raise ValueError(
+                    "firefoxLabsDescription field cannot be none "
+                    "if isFirefoxLabsOptIn is True"
+                )
+            if not data.isRollout:
+                for branch in data.branches:
+                    if branch.firefoxLabsTitle is None:
+                        raise ValueError(f"{branch.slug} missing firefoxLabsTitle")
+
+        return data
+
+    model_config = ConfigDict(
+        json_schema_extra={
+            "dependentSchemas": {
+                "isFirefoxLabsOptIn": {
+                    "if": {"properties": {"isFirefoxLabsOptIn": {"const": True}}},
+                    "then": {
+                        "properties": {
+                            "firefoxLabsTitle": {"type": "string"},
+                            "firefoxLabsDescription": {"type": "string"},
+                        },
+                        "required": ["firefoxLabsTitle", "firefoxLabsDescription"],
+                        "if": {
+                            "properties": {"isRollout": {"const": False}},
+                            "required": ["isRollout"],
+                        },
+                        "then": {
+                            "properties": {
+                                "branches": {"items": {"required": ["firefoxLabsTitle"]}}
+                            }
+                        },
```

Comment:
I would blow this up slightly so theres not so much stuff on a single line:

```suggestion
                    "if": {
                        "properties": {
                            "isFirefoxLabsOptIn": {"const": True},
                        },
                    },
                    "then": {
                        "properties": {
                            "firefoxLabsTitle": {"type": "string"},
                            "firefoxLabsDescription": {"type": "string"},
                        },
                        "required": ["firefoxLabsTitle", "firefoxLabsDescription"],
                        "if": {
                            "properties": {
                                "isRollout": {"const": False},
                            },
                            "required": ["isRollout"],
                        },
                        "then": {
                            "properties": {
                                "branches": {
                                    "items": {
                                        "required": ["firefoxLabsTitle"],
                                    },
                                },
                            },
                        },
```

(black wont reflow things onto multiple lines unless there are trailing commas)